users.js

```diff
@@ -465,6 +465,7 @@ class User {
 		/** @type {?false | string} */
 		this.semilocked = false;
 		this.namelocked = false;
+		this.battlebanned = false;
```

Comment:
Your problem is here. What is this variable supposed to be?

I would probably make it a pure string. `this.battlebanned = '';`

Reply:
(Specifically, the reason TypeScript is yelling at you is here.)

Reply:
So, first: Whenever you define a variable, you should keep in mind its contract: what it means. Here is your intended contract for this variable:

- `false`: not battle banned
- lowercase alphanumeric string: battle banned under that userid
- anything else: bug

In TypeScript, you would declare that this way:

```
/**
 * `false` if not battle banned, otherwise battle-banned userid
 * @type {string | false}
 */
this.battlebanned = false;
```

Reply:
I'm instead proposing this contract:

- `''`: not battle banned
- lowercase alphanumeric string: battle banned under that userid
- anything else: bug

You would declare that this way:

```
/** '' if not battle banned, otherwise battle-banned userid */
this.battlebanned = '';
```

Because you're only using one type now, it can be inferred, so you don't need `@type`.

Reply:
Note that your original contract is against PS's modern coding standards, anyway, see here:

https://github.com/Zarel/Pokemon-Showdown/blob/master/CONTRIBUTING.md#optionals-null-vs-undefined-vs-false

Reply:
Ah I see, I should have looked at how lock instead of namelock, I would have seen the typescript stuff lol. Also I’ll update the code standard stuff, I should have made sure the current code in the file that I was using to guide my code styling was following the contributing standards before starting lol